dataactbroker/fileRoutes.py

```diff
@@ -130,6 +130,70 @@ def get_protected_files():
+        # /v1/reviewData/
+        review_data = sess.query(Job).filter(Job.submission_id == submission_id,
+                                             Job.file_type_id.in_([6, 7]), Job.job_status_id == 4)
+        if review_data.count() > 0:
+            data = {
+                "message": "The current progress of this submission id is on /v1/reviewData/ page.",
+                "step": "5"
+            }
+            return JsonResponse.create(StatusCode.OK, data)
+
+        # /v1/validateCrossFile/
+        validate_cross_file = sess.query(Job).filter(Job.submission_id == submission_id,
+                                                     Job.file_type_id.in_([4, 5]), Job.job_type_id == 2,
+                                                     Job.number_of_errors == 0, Job.file_size.isnot(None))
+        if validate_cross_file.count() > 0:
+            data = {
+                "message": "The current progress of this submission id is on /v1/validateCrossFile/ page.",
+                "step": "3"
+            }
+            return JsonResponse.create(StatusCode.OK, data)
+
+        # /v1/generateEF/
+        generate_ef = sess.query(Job).filter(Job.submission_id == submission_id, Job.file_type_id.in_([1, 2, 3, 4, 5]),
+                                             Job.job_status_id == 2, Job.number_of_errors == 0,
+                                             Job.file_size.isnot(None))
+        if generate_ef.count() > 0:
+            data = {
+                "message": "The current progress of this submission id is on /v1/generateEF/ page.",
+                "step": "4"
+            }
+            return JsonResponse.create(StatusCode.OK, data)
+
+        # /v1/validateData/
+        validate_data = sess.query(Job).filter(Job.submission_id == submission_id,
+                                               Job.file_type_id.in_([1, 2, 3]), Job.job_type_id == 2,
+                                               Job.number_of_errors != 0, Job.file_size.isnot(None))
+        if validate_data.count() > 0:
+            data = {
+                    "message": "The current progress of this submission id is on /v1/validateData/ page.",
+                    "step": "1"
+            }
+            return JsonResponse.create(StatusCode.OK, data)
+
+        # /v1/generateFiles/
+        generate_files = sess.query(Job).filter(Job.submission_id == submission_id,
+                                                Job.file_type_id.in_([1, 2, 3]), Job.job_type_id == 2,
+                                                Job.number_of_errors == 0, Job.file_size.isnot(None))
+        if generate_files.count() > 0:
+            data = {
+                "message": "The current progress of this submission id is on /v1/generateFiles/ page.",
+                "step": "2"
+            }
+            return JsonResponse.create(StatusCode.OK, data)
+
+        data = {
+            "message": "The submission id is not found in the database.",
```

Comment:
We should probably do this check first to avoid going through all of the other queries. Right now, this is the base case so it'll return by default if all the other queries return with 0 results, so it'll go through each query first then it'll see that a submission with that ID doesn't exist. 

Also, the message should be something like "A submission with the specified ID does not exist"